discovery/client.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package discovery
+
+import (
+	"context"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/discovery/api/v1/client"
+	"github.com/nuts-foundation/nuts-node/discovery/log"
+	"time"
+)
+
+// clientUpdater is responsible for updating the presentations for the given services, at the given interval.
+// Callers should only call update().
```

Comment:
I discussed this with @gerardsn and decided it isn't worth the hassle. Making a client package causes more refactoring, leading to lots of small packages;

- Package for shared domain types (`Tag` and `Timestamp`) and another for other shared functionality (`presentationVerifier`)
- Package for storage (since `sqlStore` is shared)
- Package for shared API types

Would be fixable with a `types` package, but that ugly i.m.o. (packages should be named after the functionality they contain). So as the types and functions are internal, I considered this the lesser evil?